pkg/ccl/backupccl/restore_planning.go

```diff
@@ -1075,6 +1075,13 @@ func restorePlanHook(
 		if err != nil {
 			return nil, nil, nil, false, err
 		}
+	} else {
```

Comment:
Before the 22.1 release, I think we'll need to add a docs page that provides step by step guidance on restoring backups with the new restore syntax that were originally created with the old backup syntax. That page should be linked to this warning.

Reply:
I like the idea of a section about this, but I think it can either be inline where this link points to today, or be linked out from there.

Reply:
fwiw, It seems like inline would be a good idea. But, I also think expanding the deprecation notice and recording this in the docs — especially if workarounds are needed if a backup was taken with `BACKUP TO`, but would need to be restored with the new `RESTORE` syntax is important. When the docs issue comes through for this with the release notes, I'll prioritize into 22.1 work.

Reply:
I don't know if we need workarounds: recall RESTORE only supports restoring backups from the previous major version or newer. if 22.1 deprecates `backup to`, but can still restore backups made with it (in 21.2), then 22.2 can just drop support and we don't need to supply workarounds since any 22.1 backups _should_ be in collections anyway

Reply:
^^^ are we deprecating or merely issuing a warning for `backup to` in 22.1?

Reply:
what's the difference ?

Reply:
It will probably take all users a bit of time after the 22.1 release to refactor their code to the new syntax, so I think we should support restoring those backups that were created before they refactored their code.

My main point is, we ought to have docs that spell out the best practices for rewriting your restore command s.t. you can restore backups originally created with the old syntax. The user shouldn't try to move any SSTs around  to comply with the deprecation. And I think it's worth writing a test to ensure our best practice directions are actually correct.

Reply:
I'm still not following.

> rewriting your restore command s.t. you can restore backups originally created with the old syntax

Who would need to do this? If you made the backup with `BACKUP TO`, either in 21.2 where it was supported or in 22.1 ignoring the deprecation notice, you can just restore it with `RESTORE FROM x`, and ignore the deprecation notice; there's no need to rewrite your restore command (and I'm not sure we want you to, since the might violate some collection-assumptions if you do).

If you made you backup in 22.1 using a supported method (no deprecation notice), it is in a collection and the new syntax works, and will work in 22.2 as well.

Either way, I don't think we need to worry about new RESTORE of old BACKUP. What am I missing?

Reply:
I think I agree with David, FWIW. Our best practice could just be "load your backup with 22.1, then write a new backup using the new syntax for 22.2 support." Seems simple to understand and execute, and no additional work for...well, anyone, really, provided users don't try to upgrade two major versions within 24 hours.

Reply:
^ I like Ben's idea. 

I guess i'm concerned about a user on 22.1 that uses `BACKUP TO` who would like to restore that backup on 22.2. What should we tell them?

Reply:
> ^ I like Ben's idea.
> 
> I guess i'm concerned about a user on 22.1 that uses `BACKUP TO` who would like to restore that backup on 22.2. What should we tell them?

```The `BACKUP TO` syntax will be removed in a future release, please switch over to using `BACKUP INTO` to create a backup collection``` 😁

Reply:
...More practically, I do think we should reach out to the relevant teams to prep our users on this, but at the end of the day deprecated means deprecated to me, and we should consider 22.1 to be our transitional version.

Reply:
>recall RESTORE only supports restoring backups from the previous major version or newer

doh, didnt recall that this policy was now a thing.

Reply:
perhaps then we should make the `BACKUP TO` warning even scarrier: "your backup will not be restorable in the next major release"

Reply:
But it is restorable right?

`BACKUP TABLE foo TO 'nodelocal://1/bar' can be restored using `RESTORE FROM 'bar' IN 'nodelocal://1/';`? The `x IN y` just appends the path to `y/x` and looks for the manifest to restore.

Reply:
yeah, that happy path is restorable, but there may be other user backup file configurations that may not be (e.g. ones that use `INCREMENTAL FROM` or locality aware backups). But i'm not sure.

Reply:
> But it is restorable right?

_technically_ yes, but I'd just as soon sidestep this whole mixing old and new and just say it isn't.

Reply:
Added a slightly sterner warning, let me know what yall think.